### ISSUE TITLE
[Bug]: responseMeta.header undefined when first call api

### ISSUE THREAD

**Issue by MobGG** [appsmithorg/appsmith#28823]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

api first response in JSObject responseMeta will be
```
{
  headers: undefined,
  isExecutionSuccess: false,
  statusCode: undefined
}
```

### Steps To Reproduce

1. click button
2. check log

### Public Sample App

https://app.appsmith.com/app/responsemetabug/example-65520f7e981c6313288a33cf

### Environment

Production

### Version

cloud

**Comment by Nikhil-Nandagopal** [comment]
@MobGG it doesn't seem like your API was run and that is why you are facing this issue

**Comment by MFEC-Mob** [comment]
Excuse me can you review this issue. I updated sample app.
example api
```
https://dog.ceo/api/breed/shiba/images/random
```
code in jsObject
```	
async getShiba () {
  try {
    await getShiba.run()
    console.log('responseMeta', getShiba.responseMeta)
  } catch (e) {
    console.error(e)
  }
}
```

**Comment by Nikhil-Nandagopal** [comment]
@MFEC-Mob await will return a promise and move to the next code execution line. Can you try
getShiba.run().then((res) => {
    console.log('responseMeta', res.responseMeta)
})

**Comment by kengonline** [comment]
@Nikhil-Nandagopal Both below code should produce the same result because await will turn asynchronous code into synchronous(Not execute the next line until code `with` await statement is resolved)

`
await getShiba.run()
    console.log('responseMeta', getShiba.responseMeta)
`

`getShiba.run().then((res) => {
console.log('responseMeta', res.responseMeta)
})`

**Comment by Nikhil-Nandagopal** [comment]
@kengonline await will suspend the execution of the function called but not the calling function. So those 2 examples are not the same. 
I see an error in the example I gave, `res` is the data so it does not contain the responseMeta.
Can you try the following example

```
getShiba.run().then((res) => {
    console.log('responseMeta', getShiba.responseMeta)
})
```

**Comment by MobGG** [comment]
I updated the code. current console.log response was
```
{
  "headers": undefined,
  "isExecutionSuccess": false,
  "statusCode": undefined
}
```

**Comment by EdmilsonSantana** [comment]
Hey! I'm having the same problem. How was this problem resolved?

**Comment by MFEC-Mob** [comment]
@EdmilsonSantana 
You can use fetch to get headers.
```
const response = await fetch('https://yourapi');
const data = await response.json();
console.log('fetch headers', response.headers);
console.log('fetch status', response.status);
console.log('fetch ok', response.ok);
console.log('fetch data', data);
```

**Comment by vn7n24fzkq** [comment]
I just got the same problem and arrived here.

I found that the `data` and the  `responseMeta` updated after the program left the scope of the async function.

If you clear it before running it, you will get undefined every time.

In JSObjects
```javascript
async getShiba () {
  try {
    await getShiba.clear()
    await getShiba.run()
    console.log('responseMeta', getShiba.responseMeta)
  } catch (e) {
    console.error(e)
  }
}
```

According to `Appsmith is made with react.js`, so I guess it is a common react.js  setState & update issue.
